diff --git a/core/modules/content_moderation/src/EntityOperations.php b/core/modules/content_moderation/src/EntityOperations.php
index 52d2ca390..478d32df2 100644
--- a/core/modules/content_moderation/src/EntityOperations.php
+++ b/core/modules/content_moderation/src/EntityOperations.php
@@ -116,7 +116,8 @@ public function entityPresave(EntityInterface $entity) {
       // This entity is default if it is new, the default revision, or the
       // default revision is not published.
       $update_default_revision = $entity->isNew()
-        || $current_state->isDefaultRevisionState()
+        || (!$entity->isNew() && isset($entity->original))
+        ||  $current_state->isDefaultRevisionState()
         || !$this->moderationInfo->isDefaultRevisionPublished($entity);

       // Fire per-entity-type logic for handling the save process.
