diff --git a/docroot/core/modules/content_moderation/src/EntityOperations.php b/docroot/core/modules/content_moderation/src/EntityOperations.php
index 52d2ca39..e9b2fb07 100644
--- a/docroot/core/modules/content_moderation/src/EntityOperations.php
+++ b/docroot/core/modules/content_moderation/src/EntityOperations.php
@@ -116,8 +116,9 @@ public function entityPresave(EntityInterface $entity) {
       // This entity is default if it is new, the default revision, or the
       // default revision is not published.
       $update_default_revision = $entity->isNew()
-        || $current_state->isDefaultRevisionState()
-        || !$this->moderationInfo->isDefaultRevisionPublished($entity);
+      || (!$entity->isNew() && isset($entity->original))
+      ||  $current_state->isDefaultRevisionState()
+      || !$this->moderationInfo->isDefaultRevisionPublished($entity);
 
       // Fire per-entity-type logic for handling the save process.
       $this->entityTypeManager
