diff --git docroot/core/modules/content_moderation/src/Entity/Handler/ModerationHandler.php docroot/core/modules/content_moderation/src/Entity/Handler/ModerationHandler.php
index 243208a0..f10119a1 100644
--- docroot/core/modules/content_moderation/src/Entity/Handler/ModerationHandler.php
+++ docroot/core/modules/content_moderation/src/Entity/Handler/ModerationHandler.php
@@ -38,7 +38,6 @@ public function onPresave(ContentEntityInterface $entity, $default_revision, $pu
     // revisions which are not part of editorial updates triggered by normal
     // content changes.
     if (!$entity->isSyncing()) {
-      $entity->setNewRevision(TRUE);
       $entity->isDefaultRevision($default_revision);
     }
 
