diff --git a/mobile_app_links/src/Form/AndroidConfigForm.php b/mobile_app_links/src/Form/AndroidConfigForm.php
--- a/mobile_app_links/src/Form/AndroidConfigForm.php
+++ b/mobile_app_links/src/Form/AndroidConfigForm.php
@@ -65,6 +65,20 @@
           '#description' => $this->t('Enter one value per line.'),
           '#default_value' => $android_package['sha256_cert_fingerprints'] ?? '',
         ];
+        	
+        /* Kosovo Country package details */
+        $form['kosovo_package_name'] = [
+          '#type' => 'textfield',
+          '#title' => $this->t('Kosovo Package Name'),
+          '#default_value' => $config->get('kosovo_package_name'),
+        ];
+
+        $form['kosovo_sha256_cert_fingerprints'] = [
+          '#type' => 'textarea',
+          '#title' => $this->t('Kosovo SHA256 Certificate Fingerprints'),
+          '#description' => $this->t('Enter one value per line.'),
+          '#default_value' => $config->get('kosovo_sha256_cert_fingerprints'),
+        ];
 
         $form['android_packages'][$key]['android_package_delete'] = [
           '#type' => 'submit',
@@ -176,7 +190,12 @@
       }
       $index++;
     }
-
+    
+    /* Kosovo Package details */
+    $config->set('kosovo_package_name', $form_state->getValue('kosovo_package_name'));
+    $certificates = str_replace("\r\n", "\n", $form_state->getValue('kosovo_sha256_cert_fingerprints'));
+    $certificates = str_replace("\r", "\n", $certificates);
+    $config->set('kosovo_sha256_cert_fingerprints', $certificates);
     $config->set('android_packages', $android_packages);
     $config->save();
 
