--- docroot/modules/contrib/mobile_app_links/src/Form/AndroidConfigForm.php	2024-11-18 14:48:48.000000000 +0530
+++ androidconfigform.bkp.php	2025-10-06 18:26:32.687433817 +0530
@@ -113,6 +113,20 @@
     ];
 
     $form_state->setCached(FALSE);
+
+    $form['kosovo_package_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Kosovo Package Name'),
+      '#default_value' => $config->get('kosovo_package_name'),
+    ];
+
+    $form['kosovo_sha256_cert_fingerprints'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Kosovo SHA256 Certificate Fingerprints'),
+      '#description' => $this->t('Enter one value per line.'),
+      '#default_value' => $config->get('kosovo_sha256_cert_fingerprints'),
+    ];
+
     $form['actions']['submit'] = [
       '#type' => 'submit',
       '#value' => $this->t('Submit'),
@@ -177,6 +191,12 @@
       $index++;
     }
 
+    // Save Kosovo package details
+    $config->set('kosovo_package_name', $form_state->getValue('kosovo_package_name'));
+    $certificates = str_replace("\r\n", "\n", $form_state->getValue('kosovo_sha256_cert_fingerprints'));
+    $certificates = str_replace("\r", "\n", $certificates);
+    $config->set('kosovo_sha256_cert_fingerprints', $certificates);
+
     $config->set('android_packages', $android_packages);
     $config->save();
 
