diff --git docroot/modules/contrib/mobile_app_links/src/Form/AndroidConfigForm.php docroot/modules/contrib/mobile_app_links/src/Form/AndroidConfigForm.php
index 038ba954..789ada31 100644
--- docroot/modules/contrib/mobile_app_links/src/Form/AndroidConfigForm.php
+++ docroot/modules/contrib/mobile_app_links/src/Form/AndroidConfigForm.php
@@ -47,6 +47,20 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#default_value' => $config->get('sha256_cert_fingerprints'),
     ];
 
+    /* Kosovo Country package details */
+    $form['kosovo_package_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Kosovo Package Name'),
+      '#default_value' => $config->get('kosovo_package_name'),
+    ];
+
+    $form['kosovo_sha256_cert_fingerprints'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Kosovo SHA256 Certificate Fingerprints'),
+      '#description' => $this->t('Enter one value per line.'),
+      '#default_value' => $config->get('kosovo_sha256_cert_fingerprints'),
+    ];
+
     return $form;
   }
 
@@ -60,6 +74,12 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $certificates = str_replace("\r\n", "\n", $form_state->getValue('sha256_cert_fingerprints'));
     $certificates = str_replace("\r", "\n", $certificates);
     $config->set('sha256_cert_fingerprints', $certificates);
+
+    /* Kosovo Package details */
+    $config->set('kosovo_package_name', $form_state->getValue('kosovo_package_name'));
+    $certificates = str_replace("\r\n", "\n", $form_state->getValue('kosovo_sha256_cert_fingerprints'));
+    $certificates = str_replace("\r", "\n", $certificates);
+    $config->set('kosovo_sha256_cert_fingerprints', $certificates);
     $config->save();
 
     return parent::submitForm($form, $form_state);
